src/loggers.rs

```diff
@@ -76,14 +78,20 @@ where
     pub(crate) fn new(
         provider: &P,
         service_name: String,
+        enterprise_number: Option<String>,
```

Comment:
Nit - shouldn't this be an Option u32?

Reply:
Sticking with string as PENs can take a dotted number form as well, such as ` 1.3.6.1.4.1.32473` as per the RFC [https://www.rfc-editor.org/rfc/rfc9371.html]